src/components/diagrams/singleLineDiagram/single-line-diagram-content.tsx

```diff
@@ -100,6 +101,23 @@ const defaultBusMenuState: BusMenuState = {
     display: false,
 };
 
+// Function to apply invalid styles for sld
+function applyInvalidStyles(svgContainer: HTMLElement) {
+    // Add invalid loadflow opacity for specific classes
+    const invalidElements = svgContainer.querySelectorAll(
+        '.sld-active-power, .sld-reactive-power, .sld-voltage, .sld-angle'
```

Comment:
One minor remark: we could create a constant instead of hard coding the class names